### PR TITLE
Feature/delete node with invalid credentials

### LINKED ISSUE #286: Delete cluster where the secret is not valid anymore
Currently it is not possible to delete a cluster if you first delete the access tokens.

You can try it with this one https://beta.kubermatic.io/dc/europe-west1/cluster/7vtb9lfj9

### PR DESCRIPTION
**Pull request by lbb**
Resolves #286

### CODE REVIEW

File: provider/types.go [hunk 1]
```diff
@@ -17,6 +17,9 @@ const (
 	BareMetalCloudProvider    = "baremetal"
 )
 
+// ErrInvalidToken is raised when a given token is not valid.
+var ErrInvalidToken = fmt.Errorf("invalid token")
```
Comment (scruplelesswizard) [inline provider/types.go]: Please provide a more descriptive error:
Which token is invalid? 
What are the effects of the token being invalid?
Does the user have to take any response?

**Review by scruplelesswizard — CHANGES_REQUESTED** [review]
Please provide a more clear error message

File: provider/types.go (hunk 1 above)
Comment (lbb) [reply provider/types.go]: Please consider https://github.com/kubermatic/api/blob/4e26433d4bc36904043cd5fe72878db25a32c600/handler/cluster.go#L302

Comment (scruplelesswizard) [reply provider/types.go]: >Which token is invalid?

While there is an error string prepended to this in the calling code, we still need to be clear at all levels what is going on.
You still need to explain which token is invalid at a _bare minimum_. Also, calling it the 'cloud provider' token is insufficient, as the user may not associate 'Cloud Provider' with AWS/DO/etc...

File: handler/cluster.go [hunk 2]
```diff
@@ -290,7 +298,11 @@ func deleteClusterEndpoint(
 			}
 		}
 
-		return nil, nil
+		if len(deleteErrors) > 0 {
+			err = fmt.Errorf("Error deleting resources on the cloud provider, please manually care about the cleanup: %v", deleteErrors)
```
Comment (scruplelesswizard) [inline handler/cluster.go]: >please manually care about the cleanup

This is not valid English. You should instead use:
> please manually clean up any Storage, Nodes or Load Balancers associated with `<the cluster name>`

**Review by scruplelesswizard — CHANGES_REQUESTED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Review by scruplelesswizard — APPROVED** [review]
Lgtm

**Pull request merged**